synapse/cortex.py

```diff
@@ -253,29 +235,56 @@ def count(self, text, opts=None):
         Returns:
             (int): The number of nodes resulting from the query.
         '''
-        with self.cell.snap(user=self.user) as snap:
-            return sum((1 for n in snap.eval(text, opts=opts, user=self.user)))
+        i = 0
+        async with await self.cell.snap(user=self.user) as snap:
+            async for _ in snap.eval(text, opts=opts, user=self.user):
+                i += 1
+        return i
 
-    def eval(self, text, opts=None):
+    async def eval(self, text, opts=None):
         '''
         Evalute a storm query and yield packed nodes.
         '''
-        with self.cell.snap(user=self.user) as snap:
-            yield from snap.iterStormPodes(text, opts=opts, user=self.user)
+        MSG_QUEUE_SIZE = 1000
+        chan = asyncio.Queue(MSG_QUEUE_SIZE, loop=self.loop)
+
+        async def runEval(chan):
```

Comment:
is there namespace clashing between `chan` as a local and `chan` as a function variable? it is slightly confusing to read.